### ISSUE TITLE
[s2s examples] dataset porting

### ISSUE THREAD

**Issue by stas00** [huggingface/transformers#10044]
We need to port these to jsonlines and ideally make them part of dataset hub:
https://cdn-datasets.huggingface.co/translation/wmt_en_ro.tar.gz
https://cdn-datasets.huggingface.co/translation/wmt_en_de.tgz
https://cdn-datasets.huggingface.co/summarization/cnn_dm_v2.tgz

The problem is that nobody knows how these came to be - they are pre-processed but it's unclear how.

@sshleifer, do you by chance remember how these were created? If we put those on the new datasets hub - it'd be good to explain how these are different from normal wmt datasets. 

Also do you remember which wmtXX they came from?

Thank you!

----

The resolution is here: https://github.com/huggingface/transformers/issues/10044#issuecomment-779555741

**Comment by stas00** [comment]
Here is a script to convert these:

```
import io
import json
import re

src_lang, tgt_lang = ["en", "ro"]

for split in ["train", "val", "test"]:
    recs = []
    fout = f"{split}.json"
    with io.open(fout, "w", encoding="utf-8") as f:
        for type in ["source", "target"]:
            fin = f"{split}.{type}"
            recs.append([line.strip() for line in open(fin)])
        for src, tgt in zip(*recs):
            out = {"translation": { src_lang: src, tgt_lang: tgt } }
            x = json.dumps(out, indent=0, ensure_ascii=False)
            x = re.sub(r'\n', ' ', x, 0, re.M)
            f.write(x + "\n")
```

**Comment by sshleifer** [comment]
The short answer is I don't recall, but I have some guesses:

[en-de](https://github.com/pytorch/fairseq/blob/master/examples/translation/prepare-wmt14en2de.sh#L1)

[en-ro](https://github.com/rsennrich/wmt16-scripts/blob/master/sample/download_files.sh)

[cnn_dm](https://github.com/abisee/cnn-dailymail#option-1-download-the-processed-data)

**Comment by stas00** [comment]
That's perfect, @sshleifer! Thanks a lot!

I missed one more entry: https://cdn-datasets.huggingface.co/summarization/xsum.tar.gz

I found the main source at https://github.com/EdinburghNLP/XSum/tree/master/XSum-Dataset but not sure about what pre-processing if any went it.

Thank you!

**Comment by sshleifer** [comment]
Preprocessing: I don't actually know. The author sent me a link (in a github issue I can't find) which leads me to predict with high confidence that the preprocessing is the same as whatever is in the repo.

There is a larger *scraping wherever they are scraping is not deterministic* issue discussed in this thread https://github.com/huggingface/datasets/issues/672 (quentin and I run same code get different numbers)
which I was far too lazy to do anything more than complain about :)

**Comment by stas00** [comment]
OK, this is as good as it gets, and so I will document what you shared.

I was able to to reproduce 100% the datasets you created with the instructions you provided:
- [en-de](https://github.com/pytorch/fairseq/blob/master/examples/translation/prepare-wmt14en2de.sh#L1)
- [en-ro](https://github.com/rsennrich/wmt16-scripts/blob/master/sample/download_files.sh)

This one doesn't match.
- [cnn_dm](https://github.com/abisee/cnn-dailymail#option-1-download-the-processed-data)
the content is the same but the format is quite different after all the pre-processing steps were applied. Their results are all lower-cased and tagged with `<s></s>`, and it's word-level tokenized. Yours is just clean normal text. So you must have used a different process.

If I dig into it, it looks like your source is for sure just the original with new lines removed - in fact this is what it says in the README.md:
```
wget https://cdn-datasets.huggingface.co/summarization/cnn_dm_v2.tgz
tar -xzvf cnn_dm_v2.tgz  # empty lines removed
```
target looks like a combination of `@highlight` entries into the abstract. So little pre-processing here, but I think I sorted it out.

I'm trying to preserve these datasets you created since they are awesome for monitoring any regressions in the code, because of they avail themselves to a high bleu scores on even a small sample, so I constantly using them as a reference and I can quickly see any problems if the score drops. Thank you for creating those in first place, @sshleifer!

With all datasets or pretty much any data I make, I include the scripts that created it (or a link to where it can be found), so that it's easy to know how things came to be. Mainly doing it for myself, since I am not good at remembering such details. And I don't like researching the same thing more than once.
